tests/Unit/ControlSystem/Test_Measurements.cpp

```diff
@@ -18,70 +21,119 @@
                                  domain::ObjectLabel::A>::Excision,
                              control_system::protocols::Submeasurement>);
-
 static_assert(
     tt::assert_conforms_to_v<control_system::measurements::CharSpeed<
                                  domain::ObjectLabel::A>::Horizon,
                              control_system::protocols::Submeasurement>);
-
+static_assert(tt::assert_conforms_to_v<
+              control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+                  Excision::interpolation_target_tag<example_list>,
+              intrp::protocols::InterpolationTargetTag>);
+static_assert(tt::assert_conforms_to_v<
+              control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+                  Horizon::interpolation_target_tag<example_list>,
+              intrp::protocols::InterpolationTargetTag>);
 static_assert(
-    tt::assert_conforms_to_v<
-        control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
-            Excision::interpolation_target_tag<
-                tmpl::list<control_system::TestHelpers::ExampleControlSystem>>,
-        intrp::protocols::InterpolationTargetTag>);
-
+    not control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+        Excision::event<example_list>::factory_creatable);
 static_assert(
-    tt::assert_conforms_to_v<
-        control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
-            Horizon::interpolation_target_tag<
-                tmpl::list<control_system::TestHelpers::ExampleControlSystem>>,
-        intrp::protocols::InterpolationTargetTag>);
+    not control_system::measurements::CharSpeed<domain::ObjectLabel::A>::
+        Horizon::event<example_list>::factory_creatable);
 
 namespace {
+void test_serialization() {
+  using EventPtr = std::unique_ptr<::Event>;
+  using non_factory_creatable_event =
+      control_system::measurements::BothHorizons::FindHorizon<
+          ::domain::ObjectLabel::A>::event<example_list>;
+  using factory_creatable_event =
+      typename non_factory_creatable_event::factory_creatable_class;
+
+  register_classes_with_charm<non_factory_creatable_event,
+                              factory_creatable_event>();
+
+  const EventPtr non_factory_event =
+      std::make_unique<non_factory_creatable_event>();
+  const EventPtr factory_event = std::make_unique<factory_creatable_event>();
+
+  const EventPtr serialized_non_factory_event =
+      serialize_and_deserialize(non_factory_event);
+  const EventPtr serialized_factory_event =
+      serialize_and_deserialize(factory_event);
+
+  const auto get_typeid = [&](const EventPtr& event_ptr) {
+    const auto& event_ref = *event_ptr.get();
+    return typeid(event_ref).name();
+  };
+
+  CHECK(get_typeid(serialized_non_factory_event) ==
+        get_typeid(non_factory_event));
+  CHECK(get_typeid(serialized_factory_event) == get_typeid(factory_event));
```

Comment:
Why did you do this instead of the thing I suggested?  I don't think this is guaranteed to work.

Reply:
When I did you suggestion I got a compilation warning from clang (which we have as an error):
```
error: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Werror,-Wpotentially-evaluated-expression]
```
I googled it and it's something related to actually calling `.get()` and dereferencing the pointer inside `typeid()`. However, I was not aware that `typeid().name()` wouldn't give unique results

Reply:
Just silence the warning.  It's just clang being dumb.